src/components/Editor/CodeEditor.tsx

```diff
@@ -29,19 +33,68 @@ export default function CodeEditor(props: CodeEditorProps) {
   const dispatch = useDispatch();
   const classes = useStyles();
 
+  const doc = useSelector((state: IAppState) => state.docState.doc);
+  const client = useSelector((state: IAppState) => state.docState.client);
+  const loading = useSelector((state: IAppState) => state.docState.loading);
+  const errorMessage = useSelector((state: IAppState) => state.docState.errorMessage);
+  const peerClients = useSelector((state: IAppState) => state.peerState.peers);
+  const otherClientsCuror = useRef(new Map());
+
+  const disconnectClient = (clientId: string) => {
+    if (otherClientsCuror.current.has(clientId)) {
+      otherClientsCuror.current.get(clientId).removeCursor();
+      otherClientsCuror.current.delete(clientId);
+    }
+    dispatch(DisconnectPeer(clientId));
+  };
+
+  const connectClient = (clientId: string) => {
+    const existedClient = peerClients[clientId];
+
+    let color: string;
+    if (existedClient && existedClient.status === ConnectionStatus.Disconnected) {
+      color = existedClient.color;
+    } else {
+      color = randomColor();
+    }
+
+    const newClient = ClientCursor.of(clientId, color);
+    otherClientsCuror.current.set(clientId, newClient);
+    dispatch(AddPeer(clientId, color));
+  };
+
   useEffect(() => {
     dispatch(loadDocAction(true));
     dispatch(AttachDocAction(props.docKey));
-    // TODO we need to understand more how to use useEffect.
     // eslint-disable-next-line
-  }, []);
+  }, [props.docKey]);
 
-  const loading = useSelector((state: IAppState) => state.docState.loading);
-  const client = useSelector((state: IAppState) => state.docState.client);
-  const doc = useSelector((state: IAppState) => state.docState.doc);
-  const errorMessage = useSelector(
-    (state: IAppState) => state.docState.errorMessage,
-  );
+  useEffect(() => {
+    if (!client || !doc) {
+      return;
+    }
+
+    const unsubscribe = client.subscribe((event: any) => {
+      if (event.name === 'documents-watching-peer-changed') {
```

Comment:
'documents-watching-peer-changed' We should be able to provide this value in yorkie-js-sdk.